src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,67 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = '', health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      const index = Animal.alive.findIndex((animal) => animal === this);
+
+      Animal.alive.splice(index, 1);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden;
+
+  constructor(name, health, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  takeDamage(damage) {
+    if (!this.hidden) {
+      super.takeDamage(damage);
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  biteStrength;
+
+  constructor(name, health, biteStrength = 50) {
+    super(name, health);
+
+    this.biteStrength = biteStrength;
+  }
+
+  bite(target) {
+    if (!(target instanceof Herbivore)) {
+      return;
+    }
+
+    target.takeDamage(this.biteStrength);
```

Comment:
The `bite` method should check if the target is a Herbivore and if it is not currently hiding before applying damage. The current implementation does not check if the Herbivore is hiding, which means a Carnivore can bite a hidden Herbivore, which is against the requirements.